Add migration-tools.md (#7225)
pingcap/docs: data-migration/migration-tools.md

---
title: TiDB Ecosystem Tools Overview
summary: Learn an overview of the TiDB ecosystem tools.
---

# TiDB Ecosystem Tools Overview

TiDB provides multiple data migration tools for different scenarios such as full data migration, incremental data migration, backup and restore, and data replication.

This document introduces the user scenarios, advantages, and limitations of these tools. You can choose the right tool according to your needs.

The following diagram shows the user scenario of each migration tool.

![TiDB Migration Tools](/media/migration-tools.png)

The following table introduces the user scenarios, the supported upstreams and downstreams of migration tools.

> | Tool name | User scenario | Upstream (or the imported source file) | Downstream (or the output file) | Main advantage | Limitation |

Reviewer: ```suggestion
| Tool name | User scenario | Upstream (or the imported source file) | Downstream (or the output file) | Advantages | Limitation |
```

|:---|:---|:---|:---|:---|:---|

> |  [TiDB Data Migration (DM)](https://docs.pingcap.com/tidb-data-migration/stable/overview)| Data migration from MySQL-compatible databases to TiDB |  MySQL, MariaDB, Aurora, MySQL| TiDB   | <ul><li>A convenient and unified data migration task management tool that supports full data migration and incremental replication</li><li>Support filtering tables and operations</li><li>Support shard merge and migration</li></ul>  |  It is recommended to use DM to migrate full data with a size of less than 1 TB. |

Reviewer 2: ```suggestion
|  [TiDB Data Migration (DM)](https://docs.pingcap.com/tidb-data-migration/stable/overview)| Data migration from MySQL-compatible databases to TiDB |  MySQL, MariaDB, Aurora, MySQL| TiDB   | <ul><li>A convenient and unified data migration task management tool that supports full data migration and incremental replication</li><li>Support filtering tables and operations</li><li>Support shard merge and migration</li></ul>  | Data import speed is roughly the same as Lighting's tidb-backend, and much lower than Lighting's local-backend. So it is recommended to use DM to migrate full data with a size of less than 1 TB. |
```

> |  [TiDB Data Migration (DM)](https://docs.pingcap.com/tidb-data-migration/stable/overview)| Data migration from MySQL-compatible databases to TiDB |  MySQL, MariaDB, Aurora, MySQL| TiDB   | <ul><li>A convenient and unified data migration task management tool that supports full data migration and incremental replication</li><li>Support filtering tables and operations</li><li>Support shard merge and migration</li></ul>  | Data import speed is roughly the same as that of TiDB Lighting's TiDB-backend, and much lower than that of TiDB Lighting's Local-backend. So it is recommended to use DM to migrate full data with a size of less than 1 TB. |

Reviewer: ```suggestion
|  [TiDB Data Migration (DM)](https://docs.pingcap.com/tidb-data-migration/stable/overview)| Data migration from MySQL-compatible databases to TiDB |  MySQL, MariaDB, Aurora, MySQL| TiDB   | <ul><li>A convenient and unified data migration task management tool that supports full data migration and incremental replication</li><li>Support filtering tables and operations</li><li>Support shard merge and migration</li></ul>  | Data import speed is roughly the same as that of TiDB Lighting's TiDB-backend, and much lower than that of TiDB Lighting's Local-backend. So it is recommended to use DM to migrate full data with a size of less than 1 TiB. |
```

cluster         pingcap  Deploy a TiDB cluster for production
ctl             pingcap  TiDB controller suite
dm              pingcap  Data Migration Platform manager
dmctl           pingcap  dmctl component of Data Migration Platform
errdoc          pingcap  Document about TiDB errors
pd-recover      pingcap  PD Recover is a disaster recovery tool of PD, used to recover the PD cluster which cannot start or provide services normally
playground      pingcap  Bootstrap a local TiDB cluster for fun
tidb            pingcap  TiDB is an open source distributed HTAP database compatible with the MySQL protocol
tidb-lightning  pingcap  TiDB Lightning is a tool used for fast full import of large amounts of data into a TiDB cluster
tiup            pingcap  TiUP is a command-line component management tool that can help to download and install TiDB platform components to the local system
```

Choose the components to install:

{{< copyable "shell-regular" >}}

```shell
tiup install dumpling tidb-lightning
```

> **Note:**
>
> To install a component of a specific version, use the `tiup install <component>[:version]` command.

### Step 3. Update TiUP and its components (optional)

> It is recommended to see the update log and compatibility notes of the new version.

Reviewer 2: ```suggestion
It is recommended to see the release log and compatibility notes of the new version.
```
